File: CustomControlLibrary.WpfCore/CustomControlLibrary.WpfCore/CustomButton.cs
using System.Windows.Controls;
using System.ComponentModel;
using System;
using System.Windows;

namespace CustomControlLibrary.WpfCore
{
    public class CustomButton : Button
    {
        public CustomButton()
        {
            // The GetIsInDesignMode check and the following design-time 
            // code are optional and shown only for demonstration.
            if (DesignerProperties.GetIsInDesignMode(this))
            {
                Content = "Design mode active";
            }
        }

        // Adding a Date property to CustomButton to provide sample code for PropertyValueEditor
        public static readonly DependencyProperty DateProperty = DependencyProperty.Register(
            "Date",
            typeof(DateTime),
            typeof(CustomButton),
            new PropertyMetadata(DateTime.MinValue, OnDateChanged));

        public DateTime Date
        {
            get
            {
                return (DateTime)this.GetValue(DateProperty);
            }

            set
            {
                this.SetValue(DateProperty, value);
            }
        }

        private static void OnDateChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
        {
            CustomButton presenter = d as CustomButton;
            presenter.Date = (DateTime)e.NewValue;
        }
    }
}

File: CustomControlLibrary.WpfCore_VB/CustomControlLibrary.WpfCore/CustomButton.vb
Imports System.ComponentModel

Public Class CustomButton
    Inherits Button

    Public Sub New()
        ' The GetIsInDesignMode check and the following design-time 
        ' code are optional and shown only for demonstration.
        If DesignerProperties.GetIsInDesignMode(Me) Then
            Content = "Design mode active"
        End If
    End Sub

    ' Adding a Date property to CustomButton to provide sample code for PropertyValueEditor
    Public Shared ReadOnly DateProperty As DependencyProperty =
        DependencyProperty.Register("Date", GetType(Date),
                                    GetType(CustomButton),
                                    New PropertyMetadata(Date.MinValue, AddressOf OnDateChanged))

    Public Property [Date] As Date
        Get
            Return CDate(GetValue(DateProperty))
        End Get
        Set(value As Date)
            SetValue(DateProperty, value)
        End Set
    End Property

    Private Shared Sub OnDateChanged(d As DependencyObject, e As DependencyPropertyChangedEventArgs)
        Dim presenter = TryCast(d, CustomButton)
        presenter.Date = CDate(e.NewValue)
    End Sub

End Class
